### PR TITLE
Feat: added code theming

### LINKED ISSUE #1045: Theming: Code
### Subject

Code

### Description

Create a basic theming documentation for the `Code` component that showcases how you can style it.

Theming file: [Code theming file](https://github.com/chakra-ui/chakra-ui-docs/blob/main/content/docs/components/code/theming.mdx)

### Template Reference

For single part components, feel free to use the `Button` component theming doc as a template reference: [Button theming](https://github.com/chakra-ui/chakra-ui-docs/blob/main/content/docs/components/button/theming.mdx).

For multipart components, feel free to use the `Input` component theming doc as a template reference: [Input theming](https://github.com/chakra-ui/chakra-ui-docs/blob/main/content/docs/components/input/theming.mdx).

### PR DESCRIPTION
**Pull request by deevo-sage**
## 📝 Description

added new code component theming. (single-part component)

## ⛳️ Current behavior (updates)

Add theming tab in Code component and stuff required for sandpack embed.

## 💣 Is this a breaking change (Yes/No):

No

## 📝 Additional Information

referenced: Button theming config, badge theme from which code imports variants

### CODE REVIEW

File: content/docs/components/code/theming.mdx [hunk 1]
```diff
@@ -5,4 +5,156 @@ title: Code
...
+  variants: { outline },
+})
+```
+
+After customizing the code theme, we can import it in our theme file and add
+it in the `components` property:
+
+```jsx live=false
+import { extendTheme } from '@chakra-ui/react'
+import { codeTheme } from './components/code.ts'
```
Comment (nikolovlazar) [inline content/docs/components/code/theming.mdx:54]: Remove the `.ts` extension from the import.

File: content/docs/components/code/theming.mdx [hunk 2]
```diff
@@ -5,4 +5,156 @@ title: Code
...
+Let's assume we want to include a custom branded variant. Here's how we can do
+that:
+
+```jsx live=false
+import { defineStyle, defineStyleConfig } from '@chakra-ui/react'
+
+const brandPrimary = defineStyle({
+  background: 'orange.500',
+  color: 'white',
+  fontFamily: 'serif',
```
Comment (nikolovlazar) [inline content/docs/components/code/theming.mdx:75]: A `serif` font wouldn't make sense for the `Code` component. It should always be `mono`.

File: configs/sandpack-contents/component-theming/code.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+  <ChakraProvider theme={theme}>
+    <App />
+  </ChakraProvider>
+);`,
+  CodeTheme: `import { defineStyle, defineStyleConfig } from "@chakra-ui/styled-system"
+
+const baseStyle = defineStyle({
+  borderRadius: 0, // remove border radius
+  fontWeight: "semibold", // change the font weight to normal
+  fontFamily: "serif", // change the font family to monospaced
```
Comment (nikolovlazar) [inline configs/sandpack-contents/component-theming/code.js:61]: Def shouldn't have `serif` here.

File: configs/sandpack-contents/component-theming/code.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        </Code>
+        <Code variant="custom">
+          Themed custom code
+        </Code>
+        <Code variant="solid">
+          Themed solid code
+        </Code>
+        <Code variant="outline">
+          Themed outline code
+        </Code>
```
Comment (nikolovlazar) [inline configs/sandpack-contents/component-theming/code.js:21]: Can you redesign these? It's not a suitable look for the `Code` component.

**Review by nikolovlazar — CHANGES_REQUESTED** [review]

File: configs/sandpack-contents/component-theming/code.js (hunk 4 above)
Comment (deevo-sage) [reply configs/sandpack-contents/component-theming/code.js:21]: sure i'll add something else

File: content/docs/components/code/theming.mdx (hunk 2 above)
Comment (deevo-sage) [reply content/docs/components/code/theming.mdx:75]: by default it's mono so i though to write something else, ill switch to mono then

File: configs/sandpack-contents/component-theming/code.js [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+export default function App() {
+  const { toggleColorMode, colorMode } = useColorMode();
+  return (
+    <Box position="relative" h="100vh">
+      <SimpleGrid gap={12} p={12} columns={2}>
+        <Code>
+          const getCodeType = () => "default"
+          <br/>
+          getCodeType() // default
```
Comment (nikolovlazar) [inline configs/sandpack-contents/component-theming/code.js:13]: Let's make these single-line codes, not multi-line blocks.

File: configs/sandpack-contents/component-theming/code.js [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+  fontWeight: "semibold", // change the font weight to normal
+  fontFamily: "mono", // change the font family to monospaced
+})
+
+
+// Defining a custom variant
+const customVariant = defineStyle((props) => {
+  const { colorScheme: c } = props
+  return {
+    fontFamily: "sans-serif",
```
Comment (nikolovlazar) [inline configs/sandpack-contents/component-theming/code.js:79]: Let's keep this as `mono`.

**Review by nikolovlazar — CHANGES_REQUESTED** [review]

**Comment by nikolovlazar** [comment]
@deevo-sage I'll take over this PR. Thanks for contributing!

**Pull request merged**